crates/walrus-sdk/src/client/byte_range_read_client.rs

```diff
@@ -0,0 +1,415 @@
+// Copyright (c) Walrus Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Client for reading byte ranges from blobs.
+
+use std::time::Duration;
+
+use serde::{Deserialize, Serialize};
+use tokio::time::Instant;
+use walrus_core::{
+    BlobId,
+    Epoch,
+    SliverIndex,
+    encoding::{EncodingFactory, Primary, SliverData},
+    metadata::{BlobMetadataApi as _, VerifiedBlobMetadataWithId},
+};
+use walrus_sui::client::ReadClient;
+
+use crate::{
+    client::WalrusNodeClient,
+    error::{ClientError, ClientErrorKind, ClientResult},
+};
+
+/// Configuration for the ByteRangeReadClient.
+#[serde_with::serde_as]
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct ByteRangeReadClientConfig {
+    /// The maximum number of attempts to retrieve slivers.
+    pub max_retrieve_slivers_attempts: usize,
+    /// The timeout duration for retrieving slivers.
+    #[serde_as(as = "serde_with::DurationSeconds")]
+    #[serde(rename = "timeout_secs")]
+    pub timeout: Duration,
+}
```

Comment:
As discussed [here](https://github.com/MystenLabs/walrus/pull/2708#issuecomment-3533582022), this could be combined with the `QuiltClientConfig`.